### PR TITLE
Improve text and design of the block removal warnings

### LINKED ISSUE #52392: Update copy of warning for block removal
Looking over 6.3 beta 3 and the warning for block removal work https://github.com/WordPress/gutenberg/pull/51145, the copy/layout feels like it can be iterated upon as there's a bit of awkward spacing otherwise with items being on two lines:

Screen Shot 2023-07-06 at 12 05 07 PM

**Here's the text for Query Loop:**

> Are you sure?
> 
> Query Loop displays a list of posts or pages.
> 
> Removing this block is not advised.

**Here's the text for Post Content:**

> Are you sure?
> 
> Post Content displays the content of a post or page.
> 
> Removing this block is not advised.

To start, let's not have two separate lines as it makes the spacing a bit awkward. 

**Idea for Query Loop**

> Are you sure?
> 
> The Query Loop block displays a list of posts or pages. Removing this block will cause your posts or pages to not be displayed in this template.

**Idea for Post Content:**

> Are you sure?
> 
> The Post Content block displays the content of a post or page. Removing this block will cause the contents of your post or page to not show. 

We could also perhaps keep the "Removing this block is not advised" text depending on how much space we want to use up/how much text we want to show. Keen to get copy help!

### PR DESCRIPTION
**Pull request by felipevelzani**
## What?

It improves the text and design of the block removal warnings:

* When the user is trying to delete only one block, add the whole text in one line.
* Change the generic message from "Removing this block is not advised." to something more open, like "Only proceed to remove this block if you're absolutely sure of what you're doing."

## Why?

The goal is to improve the UX and have a clearer message/copy.

Closes #52392.

## How?

I checked how this was added technically, and "Removing this block is not advised." is added by default on all block removal warning models.

I believe that part of the reason that it was added in this more "generic" way is to be able to loop and show a list with all warnings in case the user tries to delete multiple blocks:

With that in mind, I kept a similar structure but added this text directly to its singular/plural versions.

## Testing Instructions

1. Select the "protected" block that you would like to delete ("Query Loop" block would be an example of that).
2. Click on the Delete key to see the modal — you can compare the different versions.

## Screenshots or screencast 

**One warning:**

* Before (previous text and different lines) 
image
* After (new text and using the same line)
image

**Two or more warnings:**

* Before (previous text) 
image
* After (new text)
image

### CODE REVIEW

**Comment by apeatling** [comment]
Personally I find the one line warning too long to be comfortable to read, it would be easier split onto multiple lines.

**Comment by hanneslsm** [comment]
> Only proceed to remove this block if you're absolutely sure of what you're doing.

I find this too dramatic. I'd expect such a warning when changing the PHP Version or renaming the table prefixes or something like this. The query loop can easily put into template again and revisions can be used to revert the change.

**Comment by felipevelzani** [comment]
I see your point, @apeatling — I personally have mixed feelings about adding one line instead of multiple lines and I am not really sure what would be ideal. Happy to change it as needed once we have a confirmation on what to do here.

@hanneslsm, that makes sense! How do you feel about "Only proceed to remove this block if you're sure of what you're doing." (without "absolutely")?

**Comment by hanneslsm** [comment]
> @hanneslsm, that makes sense! How do you feel about "Only proceed to remove this block if you're sure of what you're doing." (without "absolutely")?

Possible. What about something like the following - it says in a easy language what the impact is, but not as serious as "if you're sure what you're doing"?

````
Are you sure?

• Query Loop displays a list of posts or pages.
• Post Template displays each post or page in a Query Loop.

Be careful! Deleting this block affects how or if your content is displayed.
````
or
````
Be careful! Post or Page content will not be displayed after deleting.
````

**Comment by hanneslsm** [comment]
> I personally have mixed feelings about adding one line instead of multiple lines and I am not really sure what would be ideal. Happy to change it as needed once we have a confirmation on what to do here.

A rule of thumb is having 40-75 characters per line for good readability & design.

**Comment by felipevelzani** [comment]
@annezazu do you have any feedback as well on how do you think this could work here? I like @hanneslsm's suggestion above, but this would still keep in two lines and end up having the same issue.

**Comment by annezazu** [comment]
I don't think this current language matches the language we use elsewhere (in particular the "Only proceed to remove this block if you're absolutely sure of what you're doing."). I'm going to chase down a copy review on this to get some expert wordsmiths on it. I lean more towards @hanneslsm feedback either way.

**Comment by provenself** [comment]
Editorial checking in—I like @hanneslsm's suggestion as well. For readability, we could tweak a bit further, from

> Be careful! Deleting this block affects how or if your content is displayed.

to

>Be careful! Deleting this block may affect how (or if) your content is displayed.

**Comment by felipevelzani** [comment]
Thank you! Yes, @annezazu, how about adding both to the same line as you proposed initially? How do you feel about that?

I believe we need to confirm two things: The text of this window and if/when it should show in the same line.

**Comment by annezazu** [comment]
>how about adding both to the same line as you proposed initially? How do you feel about that?

Meaning something like this? Yes!

>The Post Content block displays the content of a post or page.
>Be careful! Deleting this block may affect how (or if) your content is displayed.

**Comment by hanneslsm** [comment]
Thanks for chiming in! 
I am a big advocate for precise language in modals, error messages, etc.
Therefore, two points bother me on the new sentence
> Be careful! Deleting this block may affect how (or if) your content is displayed.

1) may it? It will. I know _"may"_ sounds more polite, but let's focus on what will happen. Something will change, otherwise we won't need a dialogue.
2) the brackets indicates that the _how_ is more important than the _if_ . However, it's deleting a block. If we introduce such a hierachy, it'd go with "if (or how)".  
This can be confusing though - people might think "Why am I getting a warning dialogue (for deleting a block) if it will not delete the content anyway?"

**Comment by jameskoster** [comment]
Unless I'm missing something, deleting the content block doesn't affect _how_ content is displayed? 

Could the sentences be combined?

> Be careful! Post or page content will not be displayed if you delete this block.

**Comment by hanneslsm** [comment]
> Unless I'm missing something, deleting the content block doesn't affect how content is displayed?

You can have multiple post templates in one query loop. Deleting one doesn't mean necessarily that no content will be displayed at all. 
But it's true, _how_ is probably misleading here. I like your suggestion @jameskoster !

**Comment by jameskoster** [comment]
One potential amendment to account for the user deleting a container that includes the Content block:

> Be careful! Post or page content will not be displayed if you delete the Content block.

**Comment by donalirl** [comment]
Chiming in from the perspective of a technical writer 👋 

The line suggested by @jameskoster is closest to the ideal here because it's succinct and specific.

Wording like "may affect how (or if) your content is displayed" and "if you're absolutely sure of what you're doing" are on the vague side. As a user reading this, I wouldn't understand the specific consequence of my action. The latter could even be interpreted as condescending (if we're being sensitive about it.)

I understand the goals here are to:

* Communicate to the user the consequence of deleting the block.
* Do so in as few words as possible so the modal doesn't appear too wide.
* Avoid awkward spacing and multiple lines in the warning.

With that in mind, I propose the body text of the modal becomes:

> Deleting the **Content block** removes the page/post content from the template.

> Deleting the **Query Loop** removes the list of posts/pages from the template.

> Deleting the **Post Template** removes the blocks used to display a post.

Notes on why this wording choice:

* I removed "Be careful!" because it repeats the same meaning as the "Are you sure?" text in the heading of the modal. 
* I've intentionally bolded the block names to help users make the connection between the warning text and the block they're deleting.
* This copy eliminates the need for the generic "Removing this block is not advised" text underneath. Are these the only three blocks that show this warning? If there are more blocks, I can provide copy for those too. 

One last thing is that I would suggest something a little stronger than "Are you sure?" to encourage users to stop and pay attention. We could go with the word "Warning" but that would be too harsh. I think "Be careful!" or "Caution" are nice because they're stronger than "Are you sure?" but not as strong as "Warning".

Happy to make any further revisions if needed. cc @mtias who asked me to offer my insight here.

**Comment by donalirl** [comment]
Mockups:
Screenshot 2023-11-29 at 7 55 29 pm
Screenshot 2023-11-29 at 7 55 34 pm

**Comment by sethmclellana8c** [comment]
I like all the suggestions so far and aim to get it as concise as possible.

I suggest something beyond just defining what the user is doing.
>Deleting the Content block removes the page/post content from the template.

User: "Sure, I understand that's what I'm doing. Delete it."

Versus this statement that warns what results on the front-end.
>Post or page content will not be displayed if you delete the Content block.

User: "That's not my intended result. I don't want to do this."

**Comment by annezazu** [comment]
Thanks for chiming in, folks. To keep this moving, we need to make a final call on phrasing while keeping in mind that we can _always_ iterate. For now, let's move forward with the following:

>Be careful! Post or page content will not be displayed if you delete this block.

I'm going to bring summarize all of the other suggestions tomorrow in a comment on the original issue for safekeeping: https://github.com/WordPress/gutenberg/issues/52392 

@scruffian can you implement this?

**Comment by felipevelzani** [comment]
@annezazu, my understanding is that we would need two texts/formats for having only one block being deleted vs. multiple blocks being deleted. Isn't that the case?

By the way, I have been busy with some other projects, so I appreciate Ben or someone else finishing the work on it — I _might_ not be able to get back to this PR for at least two weeks.

**Comment by annezazu** [comment]
>my understanding is that we would need two texts/formats for having only one block being deleted vs. multiple blocks being deleted. Isn't that the case?

Yes! Let me be clearer in that case. I would just use a plural of this for now:

"Be careful! Post or page content will not be displayed if you delete these blocks."

>By the way, I have been busy with some other projects, so I appreciate Ben or someone else finishing the work on it — I might not be able to get back to this PR for at least two weeks.

❤️ Super appreciate that and want to be mindful of how much you've put in while balancing that this is something we need to iterate on. Hoping Ben or someone else can jump in and then we can see how that lands before (likely) iterating again on the language.

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 1]
```diff
@@ -52,19 +52,30 @@ export function BlockRemovalWarningModal( { rules } ) {
 			onRequestClose={ clearBlockRemovalPrompt }
 		>
 			{ blockNamesForPrompt.length === 1 ? (
-				<p>{ rules[ blockNamesForPrompt[ 0 ] ] }</p>
+				<p>
+					{ rules[ blockNamesForPrompt[ 0 ] ] }{ ' ' }
+					{ __(
+						"Only proceed to remove this block if you're absolutely sure of what you're doing."
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
						"Be careful! Post or page content will not be displayed if you delete this block."
```

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 2]
```diff
@@ -52,19 +52,30 @@ export function BlockRemovalWarningModal( { rules } ) {
...
+						style={ { listStyleType: 'disc', paddingLeft: '1rem' } }
+					>
+						{ blockNamesForPrompt.map( ( name ) => (
+							<li key={ name }>{ rules[ name ] }</li>
+						) ) }
+					</ul>
+					<p>
+						{ ' ' }
+						{ __(
+							"Only proceed to remove these blocks if you're absolutely sure of what you're doing."
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
							"Be careful! Post or page content will not be displayed if you delete these blocks.."
```

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 3]
```diff
@@ -52,19 +52,30 @@ export function BlockRemovalWarningModal( { rules } ) {
...
+						style={ { listStyleType: 'disc', paddingLeft: '1rem' } }
+					>
+						{ blockNamesForPrompt.map( ( name ) => (
+							<li key={ name }>{ rules[ name ] }</li>
+						) ) }
+					</ul>
+					<p>
+						{ ' ' }
+						{ __(
+							"Be careful! Post or page content will not be displayed if you delete these blocks.."
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
							"Be careful! Post or page content will not be displayed if you delete these blocks."
```

**Comment by scruffian** [comment]
> Be careful! Post or page content will not be displayed if you delete this block.

@annezazu is that in addition to the per block messaging?

For example if I am removing a query block the moment it says:

>Are you sure?
>    Query Loop displays a list of posts or pages.
>    Post Template displays each post or page in a Query Loop.
> Be careful! Post or page content will not be displayed if you delete these blocks.

Do we still want those additional explanations?

**Comment by annezazu** [comment]
I think we should try to remove these for now and keep the messaging as concise as possible. It's less that we need to explain what these blocks do and more that we need to explain the impact of what happens upon removal. We likely will need to iterate on this though and @donalirl's approach might be the next thing to try with some light descriptions. In playing around with this more this morning, I also agree that we should try out changing "Are you sure?" to "Be careful!" both to save space and to communicate more severity. 

That leaves us with the following for clarity:

Before:
Screenshot 2023-12-06 at 10 52 42 AM

After:

Screenshot 2023-12-06 at 10 56 57 AM

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 4]
```diff
@@ -52,19 +52,30 @@ export function BlockRemovalWarningModal( { rules } ) {
 			onRequestClose={ clearBlockRemovalPrompt }
 		>
 			{ blockNamesForPrompt.length === 1 ? (
-				<p>{ rules[ blockNamesForPrompt[ 0 ] ] }</p>
+				<p>
+					{ rules[ blockNamesForPrompt[ 0 ] ] }{ ' ' }
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion

```

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 5]
```diff
@@ -52,19 +52,29 @@ export function BlockRemovalWarningModal( { rules } ) {
...
-					) ) }
-				</ul>
+				<>
+					<ul
+						style={ { listStyleType: 'disc', paddingLeft: '1rem' } }
+					>
+						{ blockNamesForPrompt.map( ( name ) => (
+							<li key={ name }>{ rules[ name ] }</li>
+						) ) }
+					</ul>
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
```

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 6]
```diff
@@ -52,19 +52,22 @@ export function BlockRemovalWarningModal( { rules } ) {
...
-					) ) }
-				</ul>
+				<>
+					<p>
+						{ ' ' }
+						{ __(
+							"Be careful! Post or page content will not be displayed if you delete these blocks."
+						) }
+					</p>
+				</>
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
				<p>
					{ ' ' }
					{ __(
						"Be careful! Post or page content will not be displayed if you delete these blocks."
					) }
				</p>
```

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 7]
```diff
@@ -52,19 +52,20 @@ export function BlockRemovalWarningModal( { rules } ) {
 			onRequestClose={ clearBlockRemovalPrompt }
 		>
 			{ blockNamesForPrompt.length === 1 ? (
-				<p>{ rules[ blockNamesForPrompt[ 0 ] ] }</p>
+				<p>
+					{ __(
+						"Be careful! Post or page content will not be displayed if you delete this block."
+					) }{ ' ' }
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
					) }
```

File: packages/block-editor/src/components/block-removal-warning-modal/index.js [hunk 8]
```diff
@@ -52,19 +52,20 @@ export function BlockRemovalWarningModal( { rules } ) {
...
+					) }
+				</p>
 			) : (
-				<ul style={ { listStyleType: 'disc', paddingLeft: '1rem' } }>
-					{ blockNamesForPrompt.map( ( name ) => (
-						<li key={ name }>{ rules[ name ] }</li>
-					) ) }
-				</ul>
+				<p>
+					{ ' ' }
```
Comment (scruffian) [inline packages/block-editor/src/components/block-removal-warning-modal/index.js]:
```suggestion
```

**Comment by scruffian** [comment]
I ended up moving this to a new PR so I could use the `_n` function: https://github.com/WordPress/gutenberg/pull/56869

**Comment by richtabor** [comment]
Closed per https://github.com/WordPress/gutenberg/pull/58138.

**Pull request closed without merging**